Guide/file-storage.markdown

```diff
@@ -575,3 +575,244 @@ config = do
 ```
 
 [You can find a full list of `WaiParser.set...` functions on the `Network.Wai.Parse` documentation](https://hackage.haskell.org/package/wai-extra-3.1.6/docs/Network-Wai-Parse.html#v:setMaxRequestKeyLength)
+
+## Image Style Implementation
+
+Image style is a common feature in web applications. It allows you to define different styles for an image, like a thumbnail or a preview image. The image style is then applied to the image when it's viewed. One image can have multiple image styles. So on one page it would appear as a thumbnail, on another page it would appear with a 400px x 200px dimension. The original image remains as is, untouched, and whenever there is a demand for an image style then it is generated on the fly. If an image style was already generated then it is served, saving the effort of re-generating it.
+
+Let's see how we can implement that. At first, without any access control. We add an action to the `ImageStyleController` that accepts the original image path, the desired width, and height.
+
+```haskell
+-- Web/Types.hs
+-- ...
+data ImageStyleController
+    = RenderImageStyleAction { width :: !Int, height :: !Int, originalImagePath :: !Text}
+    deriving (Eq, Show, Data)
+```
+
+Wire the new route.
+
+```haskell
+-- Web/Routes.hs
+-- ...
+instance AutoRoute ImageStyleController
+```
+
+And add it to the `FrontController`
+
+```haskell
+-- Web/FrontController.hs
+
+-- Controller Imports
+-- ...
+import Web.Controller.ImageStyle
+
+instance FrontController WebApplication where
+    controllers =
+        [ startPage StartPageAction
+        -- Generator Marker
+        , parseRoute @ImageStyleController
+        -- ...
+        ]
+```
+
+Then we have our controller logic.
+
+```haskell
+-- Web/Controller/ImageStyle.hs
+
+module Web.Controller.ImageStyle where
+
+import Web.Controller.Prelude
+import IHP.ControllerSupport
+import System.Directory (doesFileExist)
+import qualified Data.Text as T
```

Comment:
In IHP we typically import `Data.Text` as `Text`, as just `T` doesn't say anything useful

```suggestion
import qualified Data.Text as Text
```